bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/RuleRegistryImpl.java

```diff
@@ -57,6 +67,14 @@ protected void addProvider(Provider<Rule> provider) {
     public synchronized void add(Rule element) {
         super.add(element);
         addIntoRuleEngine(element);
+        notify(RuleEventFactory.createRuleAddedEvent(element, SOURCE));
+    }
+
+    private void notify(Event event) {
+        if (this.eventPublisher!=null){
+            logger.debug("RuleRegistryEvent: {}", event.toString());
+            this.eventPublisher.post(event);
```

Comment:
There is already method called "postEvent" in the base class, which does the null check and exception logging. Please use this one.